pinot-core/src/main/java/org/apache/pinot/core/minion/SegmentPurger.java

```diff
@@ -230,6 +230,11 @@ public interface RecordPurgerFactory {
      * Get the {@link RecordPurger} for the given table.
      */
     RecordPurger getRecordPurger(String rawTableName);
+
+    default RecordPurger getRecordPurger(String rawTableName, PinotTaskConfig taskConfig, TableConfig tableConfig,
```

Comment:
We don't need `rawTableName` because it can be retrieved easily from the `tableConfig`?

Reply:
Removed